src/rules/noAnyRule.ts

```diff
@@ -51,16 +58,37 @@ export class Rule extends Lint.Rules.AbstractRule {
         "Type declaration of 'any' loses type-safety. Consider replacing it with a more precise type.";
 
     public apply(sourceFile: ts.SourceFile): Lint.RuleFailure[] {
-        return this.applyWithFunction(sourceFile, walk);
+        const options: Options = {
+            ignoreRestArguments: this.ruleArguments.indexOf(OPTION_IGNORE_REST_ARGS) >= 0,
+        };
+
+        return this.applyWithFunction(sourceFile, walk, options);
     }
 }
 
-function walk(ctx: Lint.WalkContext<void>) {
+interface Options {
+    ignoreRestArguments: boolean;
+}
+
+function walk(ctx: Lint.WalkContext<Options>) {
     return ts.forEachChild(ctx.sourceFile, function cb(node: ts.Node): void {
         if (node.kind === ts.SyntaxKind.AnyKeyword) {
+            if (ctx.options.ignoreRestArguments && isRestParameterArrayType(node)) {
+                return;
+            }
+
             const start = node.end - 3;
             return ctx.addFailure(start, node.end, Rule.FAILURE_STRING);
         }
+
         return ts.forEachChild(node, cb);
     });
 }
+
+function isRestParameterArrayType(anyTypeNode: ts.Node) {
+    return (
+        anyTypeNode.parent.kind === ts.SyntaxKind.ArrayType &&
+        anyTypeNode.parent.parent.kind === ts.SyntaxKind.Parameter &&
+        anyTypeNode.parent.parent.getChildAt(0).kind === ts.SyntaxKind.DotDotDotToken
```

Comment:
Nit: although it doesn't make too much of a difference right now, we generally prefer using methods such as `isArrayTypeNode` and `isParameterDeclaration` from `tsutils`. They're a little more readable and indicate to the type system that the variable is of that node's type. That can be useful if the variables have to be handled later as that type (for example, to get members that only exist on that type).